iothub/device/src/Transport/ProtocolRoutingDelegatingHandler.cs

```diff
@@ -21,10 +21,10 @@ internal class ProtocolRoutingDelegatingHandler : DefaultDelegatingHandler
         /// <summary>
         /// After we've verified that we could open the transport for any operation, we will stop attempting others in the list.
         /// </summary>
-        private bool _transportSelectionComplete = false;
-        private int _nextTransportIndex = 0;
+        private bool _transportSelectionComplete;
+        private int _nextTransportIndex;
 
-        private SemaphoreSlim _handlerLock = new SemaphoreSlim(1, 1);
+        private readonly SemaphoreSlim _handlerLock = new SemaphoreSlim(1, 1);
```

Comment:
same comment that drwill had added above -> when this class is disposed, ideally we should dispose this semaphore and set it to null

Reply:
yap, I will go over these classes and implement dispose on all of them.